pkg/stanza/adapter/converter.go

```diff
@@ -71,13 +72,47 @@ type Converter struct {
 	// when Stop() is called.
 	wg sync.WaitGroup
 
-	logger *zap.Logger
+	logger      *zap.Logger
+	buildInfo   component.BuildInfo
+	packageName string
 }
 
 type converterOption interface {
 	apply(*Converter)
 }
 
+func withPackageName(name string) converterOption {
+	return packageNameOption{name}
+}
```

Comment:
The interface pattern allows this to be an opt-in solution, but feels kinda magical.  I'm not sold on this idea, but it does provide the value without any breaking changes and I think we do need the component to provide the value since the ID is not the same as the package name.